cfme/configure/access_control/__init__.py

```diff
@@ -1290,6 +1290,40 @@ class TenantQuotaForm(View):
     vm_txt = Input(id='id_vms_allocated')
     template_txt = Input(id='id_templates_allocated')
 
+    @staticmethod
+    def check_fill_request_is_same_as_current_form(request, current):
+        """
+        The purpose of this method is to compare values that actaully
+        are in the form right now with the values that are requested by a test.
+        We do it because widgetastic's ``View.fill`` method returns wrong value.
+        It returns ``True`` even if no values in form are to be changed.
+        As a result, ``Save`` button on ``TenantQuotaView`` cannot be clicked.
+        That's the reason why the original ``fill`` method is overloaded in this class.
+
+        Args:
+            request: A dictionary in form that is passed to ``fill`` method.
+            current: A dictionary in form that is returned by ``read`` method.
+        """
+        for request_key, request_value in request.items():
+            if request_key.endswith('_cb') and bool(request_value) != current[request_key]:
+                return False
+            if request_key.endswith('_txt'):
+                if request_value is None and current[request_key] == '':
+                    pass
+                elif str(request_value) == current[request_key]:
+                    pass
+                else:
+                    return False
+        return True
```

Comment:
**REQUIRED**
I don't think we should workaround the issue such way. The real reason should be found. Why does fill method return a wrong value?
сс @mfalesni

Reply:
@quarckster Honestly I did not think it's a workaround. I thought that fill is sometimes so specific that you need to overload it to suit your needs. But if we find more universal solution for this, I am of course happy,

Reply:
We shouldn't be using the suffixes in first place :stuck_out_tongue: 
If it returns True even if nothing was changed, can you look in the logs which widget returned True so fill returned true?

Reply:
@mfalesni Here is relevant piece of [cfme.log](https://up1-vrutkovs.int.open.paas.redhat.com/#0sZrfyiIxYkRQ7xUVMLoOg). If fails right after this, because of course no success flash message is not displayed if Save button was not clicked. I mean it probably was clicked, but it did nothing since it is not active.

Reply:
@mfalesni What do you think about it? Should we merge this and consider it as a WA? Or can you create an actual fix? Or maybe point me in the right direction? I am asking because it would be really great if the fix made it to the next downstream-stable.

Reply:
@jan-zmeskal I cannot reproduce the issue, I ran tenant_quota_test with `set_quota` code:
```python
def set_quota(self, **kwargs):
    """ Sets tenant quotas """
    view = navigate_to(self, 'ManageQuotas', wait_for_view=True)
    changed = view.form.fill({'cpu_cb': kwargs.get('cpu_cb'),
                              'cpu_txt': kwargs.get('cpu'),
                              'memory_cb': kwargs.get('memory_cb'),
                              'memory_txt': kwargs.get('memory'),
                              'storage_cb': kwargs.get('storage_cb'),
                              'storage_txt': kwargs.get('storage'),
                              'vm_cb': kwargs.get('vm_cb'),
                              'vm_txt': kwargs.get('vm'),
                              'template_cb': kwargs.get('template_cb'),
                              'template_txt': kwargs.get('template')})
    if changed:
        view.save_button.click()
        view = self.create_view(DetailsTenantView)
        view.flash.assert_success_message('Quotas for {} "{}" were saved'.format(
            self.obj_type, self.name))
    else:
        view.cancel_button.click()
        view = self.create_view(DetailsTenantView)
        view.flash.assert_success_message('Manage quotas for {} "{}" was cancelled by the user'
            .format(self.obj_type, self.name))
    assert view.is_displayed
```
I didn't changed the view at all and it works fine.

Reply:
@quarckster Try running it on an appliance where all the values that are going to be set by test are manually set before.

Reply:
@mfalesni, @quarckster Sorry to be bothering you again, but I need your ideas on this. If you want me to rework this, then ok, but at least give me some pointers. However right now `view.form.fill` returns `True` even if the desired values were in the form even before filling and that's a problem. So I cannot just leave it as it is.

Reply:
So I pointed out @jan-zmeskal fills integers into inputs, which breaks the fill logic. (2 != '2' -> fill it, and it works because we convert all fills into strings)